cranelift/codegen/src/isa/x64/lower.rs

```diff
@@ -747,6 +747,29 @@ fn lower_insn_to_regs<C: LowerCtx<I = Inst>>(
             }
         }
 
+        Opcode::Umulhi | Opcode::Smulhi => {
+            let input_ty = ctx.input_ty(insn, 0);
+            let size = input_ty.bytes() as u8;
+
+            let lhs = input_to_reg(ctx, inputs[0]);
+            let rhs = input_to_reg_mem(ctx, inputs[1]);
+            let dst = output_to_reg(ctx, outputs[0]);
+
+            // Move lhs in %rax.
+            ctx.emit(Inst::gen_move(
+                Writable::from_reg(regs::rax()),
```

Comment:
Hmm, `gen_move` has its destination on the left?  That feels inconsistent with the rest of the x64 stuff, which I thought had destinations as the rightmost argument.  But maybe that's because `gen_move` is from the applies-to-all-targets `trait Inst`, right?  I guess the use of `Writable` guarantees we're safe here.

Reply:
Indeed, `gen_move` is on the target-independent trait. Maybe we can change this at some point, if the aarch64 backend agrees with the x64 convention?

Reply:
Well, aarch64 has destinations on the left.  So it's never going to agree with x64 (the AT&T conventions, at least.).